File: tests/ackley/model/ackley.cpp
#include "ackley.h"
#include <math.h>

double ackley(double *x)
{
   double a = 20, b = .2, c = 2.*M_PI, s1 = 0., s2 = 0.;

   for (int i = 0; i < NDIMS; ++i) {
       s1 += x[i]*x[i];
       s2 += cos(c*x[i]);
   }

   return -a*exp(-b*sqrt(s1/NDIMS)) - exp(s2/NDIMS) + a + exp(1.);
}

File: examples/python/ackley/model/ackley.py
#!/usr/bin/env python
import math

def ackley(x):
 a = 20.0
 b = 0.2
 c = 2.*math.pi
 for j in range(70):
  s1 = 0.0
  s2 = 0.0
  for i in range(4):
   s1 += x.getParameter(i)*x.getParameter(i)
   s2 += math.cos(c*x.getParameter(i))

 result = -(-a*math.exp(-b*math.sqrt(s1/4)) - math.exp(s2/4) + a + math.exp(1.))
 x.addResult(result)